Dodge.js

```diff
@@ -0,0 +1,179 @@
+import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
+import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
+import AxisPad from 'react-native-axis-pad';
+////this looks interesting for the attacker placing/throwing projectiles:
+//https://reactnative.dev/docs/panresponder 
+//https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
+var ents = {}
+
+const generateUniqueId = require('generate-unique-id');
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+
+
+//adds an entity given its type, if it's a Grenade set a timer for explosion
+function AddEntity(entity){
+let id = generateUniqueId()
+ents[id] = new entity(id)
+
+if(entity == Grenade){
+  setTimeout(() => { Explode(id) }, 2000)
+}
+}
+
+//deletes a single entity given that entities unique id
+function DeleteEntity(id){
+delete(ents[id])
+}
+
+//blows up the grenade
+function Explode(id){
+  //spawns shrapnel traveling from where the grenade blew up
+  InitialShrapnel(ents[id].x,ents[id].y)
+  //delete the original projectile
+  DeleteEntity(id)
+}
+
+//runs 60 times a second updating each entitiy
+function Gameloop(entities, { touches }, )  {
+  let toDelete = []
+  for (let id in entities) {
+      entities[id].update()
+  }
+  
+return entities
+}
+
+//creates a square component
+function CreateBrick(props) {
+return(
+  //Having a dynamic call as in props.Image produces the error
+  //Cannot find module './assets/Brick.jpeg'; which doesn't seem right
+    <Image source = {require("./assets/Brick.jpeg")}
+      style = {{        
+      position: 'absolute',
+      width: props.width,
+      height: props.height,
+      left: props.x,
+      top: props.y,
+      backgroundColor: props.backgroundColor,}}/>
+
+);
+}
+/*
+//creates a square component
+function CreateBrick(props){
+  return (
+     <View style={styles.container}>
+      <Image
+        style={ INSERT STYLE HERE }
+        source={require('@expo/snack-static/react-native-logo.png')}
+      />
+    
+    <View
+      style={{
+        position: 'absolute',
+        width: props.width,
+        height: props.height,
+        left: props.x,
+        top: props.y,
+        backgroundColor: props.backgroundColor,
+      }}
+    />
+  );
+}
+*/
+//creates a circle component
+function CreateCircle(props){
+  return (
+
+    <Image
+      source = {require("./assets/Baseball.png")}
+      style={{
+        position: 'absolute',
+        borderRadius: props.borderRadius,
+        width: props.width,
+        height: props.height,
+        left: props.x,
+        top: props.y,
+        backgroundColor: props.backgroundColor,
+      }}
+    />
+  );
+}
+
+  //creates one shrapnel in each of the cardinal directions
+  function InitialShrapnel(x, y){
```

Comment:
Maybe this should be moved into the Grenade class? (you'd have to make some changes above first)

Partly just how you think about it, i.e. is it the world, or the grenade that is "responsible" for the shrapnel.
But probably, if there's only one class that's ever likely to use a function, that function should live inside that class.